src/rgw/rgw_crypt.cc

```diff
@@ -1161,26 +1257,58 @@ int rgw_s3_prepare_decrypt(struct req_state* s,
     const char *req_cust_alg =
         s->info.env->get("HTTP_X_AMZ_SERVER_SIDE_ENCRYPTION_CUSTOMER_ALGORITHM", NULL);
 
-    if ((nullptr == req_cust_alg) || (strcmp(req_cust_alg, "AES256") != 0)) {
-      ldout(s->cct, 5) << "ERROR: Invalid value for header "
+    if (nullptr == req_cust_alg)  {
+      ldout(s->cct, 5) << "ERROR: Get encryption object not provide "
                        << "x-amz-server-side-encryption-customer-algorithm"
                        << dendl;
-      return -ERR_INVALID_REQUEST;
+      s->err.message =
+        "Requests specifying Server Side Encryption with Customer provided keys must provide a valid encryption algorithm.";
+      return -EINVAL;
+    } else if (strcmp(req_cust_alg, "AES256") != 0) {
+      ldout(s->cct, 5) << "ERROR: Get encryption object not provide a valid encryption algorithm." << dendl;
+      s->err.message = "The Encryption request you specified is not valid. Supported value: AES256.";
```

Comment:
`The requested encryption algorithm is not valid, must be AES256.`

Reply:
OK